### PR TITLE
Server

### PR DESCRIPTION
**Pull request by MauriceNDS**
**Server part added !** 

Now we can connect to the server and send requests from multiple clients, enjoy !

### CODE REVIEW

File: includes/ft_irc.hpp [hunk 1]
```diff
@@ -4,6 +4,10 @@
 #include <string>
 #include <map>
 #include <vector>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <poll.h>
```
Comment (SimonCROS) [inline includes/ft_irc.hpp]: Move includes to Server.hpp if needed or Server.cpp else.

File: modules/api/command/CommandManager.cpp [hunk 2]
```diff
@@ -14,8 +15,10 @@ void CommandManager::process(MessageEvent& event) {
 	std::getline(ss, command, ' ');
 
 	it = cspecs.find(command);
-	if (it == cspecs.end())
-		throw CommandNotFoundException();
```
Comment (SimonCROS) [inline modules/api/command/CommandManager.cpp:18]: `CommandNotFoundException` now unused, file can be deleted (check for ohter exceptions if if they are used).

File: modules/server/Server.cpp [hunk 3]
```diff
@@ -4,34 +4,137 @@
...
+
+	// Can be protected -- Set socket to be nonblocking
+	ioctl(serverSocket.fd, FIONBIO, (char *)&opt);
+
+	addConnection(serverSocket);
+
+
+	connectionConfig.sin_family = AF_INET;
+	connectionConfig.sin_addr.s_addr = INADDR_ANY;
+	connectionConfig.sin_port = htons(stoi(this->port));
```
Comment (SimonCROS) [inline modules/server/Server.cpp]: `stoi`: C++11

File: modules/server/Server.cpp [hunk 4]
```diff
@@ -4,34 +4,137 @@
...
+	connections.push_back(newConnect);
+	return newConnect;
 }
 
-User *Server::simulate_join(Connection *connection, string name) {
-	User *u = new User(connection);
-	u->setNickName(name);
-	Irc::getInstance().addUser(u);
-	return u;
+Server::Server(const string& name, const string& port, const string& password) : name(name), port(port), password(password) {
```
Comment (SimonCROS) [inline modules/server/Server.cpp]: Change `port` from `const string&` to `int` type.
```suggestion
Server::Server(const string& name, int port, const string& password) : name(name), port(port), password(password) {
```

File: includes/modules/server/Server.hpp [hunk 5]
```diff
@@ -5,27 +5,33 @@
...
 #include "api/Connection.hpp"
+#include <unistd.h>
 
 class Server : public CommandSender {
 private:
 	string name;
-	map<int, Connection *> connections;
+	vector<Connection *> connections;
+	struct sockaddr_in connectionConfig;
+	const string port;
```
Comment (SimonCROS) [inline includes/modules/server/Server.hpp]: Change `port` from `const string` to `const int` type.
```suggestion
	const int port;
```

File: includes/modules/server/Server.hpp [hunk 6]
```diff
@@ -5,27 +5,33 @@
...
-	Connection *simulate_connect(int socket);
-	User *simulate_join(Connection *connection, string name);
+	void incomingConnection();
+	void incomingRequest(size_t index);
+	void closeConnection(size_t	index);
+	Connection *addConnection(const struct pollfd &connection);
 
 public:
-	Server(const string& name) : name(name) {}
+	Server(const string& name, const string& port, const string& password);
```
Comment (SimonCROS) [inline includes/modules/server/Server.hpp]: Change `port` from `const string&` to `int` type.
```suggestion
	Server(const string& name, int port, const string& password);
```

**Review by SimonCROS — CHANGES_REQUESTED** [review]

File: modules/server/Server.cpp [hunk 7]
```diff
@@ -4,34 +4,137 @@
 
 #include <fstream>
 
-Connection *Server::simulate_connect(int socket) {
-	Connection *connection = new Connection(socket, sockaddr());
-	connections[socket] = connection;
-	return connection;
+#define MAX_BUFFER_LENGTH 4096
```
Comment (Owllyy) [inline modules/server/Server.cpp]:
```suggestion
#define MAX_BUFFER_LENGTH 512
```

"these messages SHALL NOT exceed 512 characters in length" RFC 2812

**Review by Owllyy — CHANGES_REQUESTED** [review]

**Review by Owllyy — APPROVED** [review]

File: modules/server/Server.cpp [hunk 8]
```diff
@@ -4,18 +4,20 @@
...
+#define MAX_BUFFER_LENGTH 512
 
 Connection *Server::addConnection(const struct pollfd &connection) {
 	Connection *newConnect = new Connection(connection);
 	connections.push_back(newConnect);
 	return newConnect;
 }
 
-Server::Server(const string& name, const string& port, const string& password) : name(name), port(port), password(password) {
+Server::Server(const string& name, const string& port, const string& password) : name(name), password(password) {
```
Comment (SimonCROS) [inline modules/server/Server.cpp]:
```suggestion
Server::Server(const string& name, const int port, const string& password) : name(name), port(port), password(password) {
```

File: includes/modules/server/Server.hpp [hunk 9]
```diff
@@ -5,14 +5,19 @@
...
+#include <netinet/in.h>
+#include <poll.h>
 
 class Server : public CommandSender {
 private:
 	string name;
 	vector<Connection *> connections;
 	struct sockaddr_in connectionConfig;
-	const string port;
+	int port;
```
Comment (SimonCROS) [inline includes/modules/server/Server.hpp]:
```suggestion
	const int port;
```

**Review by SimonCROS — CHANGES_REQUESTED** [review]

**Review by SimonCROS — APPROVED** [review]

**Pull request merged**